pkg/daemon/update.go

```diff
@@ -556,6 +519,73 @@ func (dn *Daemon) updateOnClusterBuild(oldConfig, newConfig *mcfgv1.MachineConfi
 	return dn.reboot(fmt.Sprintf("Node will reboot into image %s / MachineConfig %s", newImage, newConfigName))
 }
 
```

Comment:
thought: This may be more involved

I have a feeling that the revert process might be more involved and may require us to go through the `updateOnClusterBuild` method, but let's try it this way first to see if it works.